src/mesh/replicated_mesh.C

```diff
@@ -1428,7 +1428,8 @@ void ReplicatedMesh::stitching_helper (const ReplicatedMesh * other_mesh,
         }
     }
 
-  this->prepare_for_use( /*skip_renumber_nodes_and_elements= */ false, skip_find_neighbors);
+  this->allow_find_neighbors(!skip_find_neighbors);
```

Comment:
And here the same bug as https://github.com/libMesh/libmesh/pull/2473/files#r435257295 is being made *manually*: when we call ReplicatedMesh::stitching_helper now with skip_find_neighbors, instead of making the call slightly cheaper, it breaks the Mesh.  This was being covered up in CI before because the use cases were always being followed with *another* old-style prepare_for_use call that inadvertently permanently restored allow_find_neighbors(), but if code paths try to use the non-deprecated prepare_for_use overload then things simply stay broken.